code/datums/helper_datums/construction_datum.dm

```diff
@@ -31,17 +31,47 @@
 	if(valid_step)
 		if(custom_action(valid_step, used_atom, user))
 			next_step()
-			return 1
-	return 0
+			return TRUE
+	return FALSE
 
 /datum/construction/proc/is_right_key(atom/used_atom) // returns current step num if used_atom is of the right type.
 	var/list/L = steps[steps.len]
 	if(istype(used_atom, L["key"]))
 		return steps.len
-	return 0
+	return FALSE
 
 /datum/construction/proc/custom_action(step, used_atom, user)

```

Comment:
the "pcontruction" datum was dumped, and it's main type, "reversible2", was ported to the main datum.